src/main.cpp

```diff
@@ -510,19 +536,19 @@ void tick_all_interfaces(const SysTick_s &current_system_tick)
             int(fsm.get_state()),
             buzzer.buzzer_is_on(),
             drivetrain.drivetrain_error_occured(),
-            torque_controller_mux.getTorqueLimit(),
+            torque_controller_mux.get_tc_mux_status().current_torque_limit_enum,
             ams_interface.get_filtered_min_cell_voltage(),
             a1.get().conversions[MCU15_GLV_SENSE_CHANNEL],
-            static_cast<int>(torque_controller_mux.activeController()->get_launch_state()),
+            static_cast<int>(torque_controller_mux.get_tc_mux_status().current_controller_mode_),
             dashboard.getDialMode());
 
         main_ecu.tick(
             static_cast<int>(fsm.get_state()),
             drivetrain.drivetrain_error_occured(),
             safety_system.get_software_is_ok(),
-            static_cast<int>(torque_controller_mux.getDriveMode()),
-            static_cast<int>(torque_controller_mux.getTorqueLimit()),
-            torque_controller_mux.getMaxTorque(),
+            static_cast<int>(torque_controller_mux.get_tc_mux_status().current_controller_mode_),
+            static_cast<int>(torque_controller_mux.get_tc_mux_status().current_torque_limit_enum),
+            torque_controller_mux.get_tc_mux_status().current_torque_limit_value,
```

Comment:
i can instead pass in the tc mux status struct instead of the tc mux itself, as that way it relies upon having just the shared data types and doesnt have to also depend / include the tc mux implementation as interfaces shouldnt rely upon specific systems.